src/Compiler/TypeToPhpDoc.php

```diff
@@ -21,6 +22,10 @@ public function __construct(TypeStringResolver $typeStringResolver)
 
     public function toPhpDocString(Type $type): string
     {
```

Comment:
What does this? When it appears? Can we add some test?

Reply:
When you assign $this to variable it generated invalid phpdoc because that type is outputted as `$this(\Some\Class)`

Reply:
I necountered it in our code where we passed $this to be able to call method form $this in template

Reply:
I saw that before, but somehow I always change the code to get Some\Class. So can you add test case?